### PR TITLE
Use async actions for system tray actions

### LINKED ISSUE #5349: Debug why nodes can't be created from `ockam_app`
### Current behavior

When trying to create a node from the `ockam_app` crate it fails, and the node appears as DOWN (uninitialized).

### Desired behavior

We want to use the [`initialize_node_if_default`](https://github.com/build-trust/ockam/blob/9ac0a184f24e0479add5cb733cdecaf2d97d5ab4/implementations/rust/ockam/ockam_command/src/node/mod.rs#L85) to initialize a default node from the `ockam_app` crate.

Figure out why the node is not being initialized.

### LINKED ISSUE #5358: Investigate why Reset in the desktop tray app creates a new tray icon
# Observed behavior

We're building a Desktop System Tray app, see https://github.com/build-trust/ockam/issues/5228

When you Enroll and Reset in the app a second tray icon appears:

https://github.com/build-trust/ockam/assets/159583/4297a3b4-db3e-436b-b1f0-f8c62a2b680a

# Steps to reproduce

```
cd implementations/typescript
pnpm install
pnpm build

cd ../../implementations/rust/ockam/ockam_app
cargo tauri dev
```

# Expected behavior

New icon should not appear.

# Source Code

The code to investigate is here

- https://github.com/build-trust/ockam/blob/develop/implementations/rust/ockam/ockam_app/src/enroll/reset.rs
- https://github.com/build-trust/ockam/blob/develop/implementations/rust/ockam/ockam_app
- https://github.com/build-trust/ockam/blob/develop/implementations/typescript/ockam/ockam_app

### LINKED ISSUE #5360: Investigate how to remove the blocking beachball during Enroll in the desktop app
# Observed behavior

We're building a Desktop System Tray app for Ockam, see https://github.com/build-trust/ockam/issues/5228

See the 8 second mark of the below video

https://github.com/build-trust/ockam/assets/159583/4297a3b4-db3e-436b-b1f0-f8c62a2b680a

During Enroll something blocks the UI, we don't want that.
This is somewhat related to this comment. https://github.com/build-trust/ockam/issues/5241#issuecomment-1636937919

# Steps to reproduce

```
cd implementations/typescript
pnpm install
pnpm build

cd ../../implementations/rust/ockam/ockam_app
cargo tauri dev
```

# Expected behavior

Beachball should not appear. The UP should remain interactive during Enroll.

# Source Code

The code to investigate is here

- https://github.com/build-trust/ockam/blob/develop/implementations/rust/ockam/ockam_app
- https://github.com/build-trust/ockam/blob/develop/implementations/typescript/ockam/ockam_app

### PR DESCRIPTION
**Pull request by etorreborre**
This PR introduces some application state which:

 - holds the global command options for operations requiring at data on disk
 - holds a `Context` to run async actions (and makes sure that the tauri runtime is the same as the context runtime)
 - holds a `NodeManager` so that operations list creating outlets or listing them can be done directly without having to send requests to a background node

Additionally it solves:

 - #5360  -> this is solved by running async actions
 - #5358  -> that's because we were trying to start a background node with `ockam` as the executable (and in that case this is the desktop app)
 - #5349 -> we don't try to create them anymore

There is one ongoing issue with this PR which is some sort of loop on checking the sessions after the enrollment, which I am still trying to debug. **EDIT** the sessions issue is fixed.

### CODE REVIEW

File: implementations/rust/ockam/ockam_app/src/lib.rs [hunk 1]
```diff
@@ -1,24 +1,25 @@
...
 mod error;
 mod quit;
 mod tcp;
 
 #[cfg_attr(mobile, tauri::mobile_entry_point)]
 pub fn run() {
-    let args = GlobalArgs::default().set_quiet();
-    let options = CommandGlobalOpts::new(args);
-    let options_clone = options.clone();
+    println!("making the app state");
```
Comment (shanesveller) [inline implementations/rust/ockam/ockam_app/src/lib.rs]: Was this one a leftover debugging statement? If not, should this become a `tracing` call instead? We'd have to initialize a subscriber though, and that may conflict with the intent or implementation details of our Tauri logging plugin.

File: implementations/rust/ockam/ockam_app/src/app/app_state.rs [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+}
+
+impl AppState {
+    /// Create a new AppState
+    pub fn new() -> AppState {
+        let options = CommandGlobalOpts::new(GlobalArgs::default().set_quiet());
+        let (context, mut executor) = NodeBuilder::new().no_logging().build();
+        let context = Arc::new(context);
+
+        // from now on we use the same runtime everywhere we need to run an async action
```
Comment (shanesveller) [inline implementations/rust/ockam/ockam_app/src/app/app_state.rs:50]: 💯

File: implementations/rust/ockam/ockam_app/src/app/app_state.rs [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+        }
+    }
+}
+
+/// Make a node manager with a default node called "default"
+async fn make_node_manager(
+    ctx: Arc<Context>,
+    opts: CommandGlobalOpts,
+) -> miette::Result<NodeManager> {
+    let node_name = "default";
```
Comment (shanesveller) [inline implementations/rust/ockam/ockam_app/src/app/app_state.rs:108]: Will this show up in commands like `ockam node list`? If so, should we consider tweaking the name to more clearly associate it with the app?

Comment (etorreborre) [reply implementations/rust/ockam/ockam_app/src/app/app_state.rs:108]: Yes that would be a good idea (for a subsequent PR)

**Review by adrianbenavides — APPROVED** [review]

**Pull request merged**